veps/sig-compute/hypervisor-abstraction.md

```diff
@@ -0,0 +1,559 @@
+# VEP #97: Hypervisor Abstraction Layer
+
+## Release Signoff Checklist
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [x] (R) Enhancement issue created, which links to VEP dir in [kubevirt/enhancements]
+- [ ] (R) Target version is explicitly mentioned and approved
+- [ ] (R) Graduation criteria filled
+
+## Overview
+
+Introduce a Hypervisor Abstraction Layer that lets KubeVirt plug in multiple hypervisor backends through a consistent contract, while keeping today's KVM-first behavior unchanged. To start, we scope the contract to device selection, domain tweaks, validation and mutation so as to provide a flexible base that can evolve without introducing a major refactor. 
+
+## Motivation
+
+- KubeVirt currently hard-codes KVM/QEMU assumptions throughout virt-launcher, virt-handler, virt-controller, and node preparation scripts.
+- Platform teams that rely on alternative accelerators or hypervisor stacks face invasive forks to replace `/dev/kvm`, libvirt domain types, or resource scheduling hints.
+- A scoped abstraction keeps the project approachable for contributors while unlocking new hardware backends.
+
+## Goals
+
+- Document the cluster-wide hypervisor configuration and per-component extension points (defaults, converter, webhooks, node labeller) so downstream implementations can extend behavior without invasive changes to existing components.
+- Resolve the active hypervisor early and feed it through the virt-launcher converter so hypervisor-specific behavior stays localized.
+- Support both admission-time validation and mutation so administrators can enforce guardrails while still customizing VMIs for a given hypervisor.
+- Let components request allocatable device resources declared by the hypervisor configuration, avoiding new scheduling primitives.
+- Make it simple for downstreams to implement new hypervisors by following a documented contract.
+
+## Non Goals
+
+- Deliver a full implementation of any specific new hypervisor backend.
+- Redesign the VirtualMachineInstance API schema beyond additive fields.
+- Replace existing Hyper-V enlightenment features or other architecture-specific helpers.
+- Mandate new observability requirements; telemetry hooks remain optional.
+
+## Definition of Users
+
+- Cluster administrators who need to bootstrap KubeVirt on hardware that exposes alternative virtualization devices.
+- Platform vendors integrating proprietary or emerging hypervisor stacks with KubeVirt.
+- Upstream contributors maintaining virt-launcher, virt-controller, and virt-handler.
+
+## User Stories
+
+1. As a cluster administrator, I can declare a non-KVM hypervisor as the cluster default, and VMI pods schedule only on nodes that expose its required devices.
```

Comment:
I am not sure about this user story. Saying that we have a default, suggests that someone (VM  owner?) can override it and set another value.

Reply:
Sorry if I wasn't clear - I don't fully understand what a "default" means here. I'd prefer
```suggestion
1. As a cluster administrator, I can I would like to deploy KubeVirt on a cluster with non-KVM hypervisor nodes, and have non-KVM VMs schedule only on nodes that expose its required devices.
```